src/gui/raster/qgsresamplingutils.cpp

```diff
@@ -0,0 +1,186 @@
+/***************************************************************************
+  qgsresamplingutils.cpp
+  --------------------
+      begin                : 12/06/2020
+      copyright            : (C) 2020 Even Rouault
+      email                : even.rouault at spatialys.com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsrasterlayer.h"
+#include "qgsrasterdataprovider.h"
+#include "qgsrasterresampler.h"
+#include "qgsrasterresamplefilter.h"
+#include "qgsresamplingutils.h"
+#include "qgsbilinearrasterresampler.h"
+#include "qgscubicrasterresampler.h"
+
+#include <QObject>
+#include <QComboBox>
+#include <QDoubleSpinBox>
+#include <QCheckBox>
+
+QgsResamplingUtils::QgsResamplingUtils() = default;
+
+void QgsResamplingUtils::initWidgets( QgsRasterLayer *rasterLayer,
+                                      QComboBox  *zoomedInResamplingComboBox,
+                                      QComboBox *zoomedOutResamplingComboBox,
+                                      QDoubleSpinBox *maximumOversamplingSpinBox,
+                                      QCheckBox *cbEarlyResampling )
+{
+  mRasterLayer = rasterLayer;
+  mZoomedInResamplingComboBox = zoomedInResamplingComboBox;
+  mZoomedOutResamplingComboBox = zoomedOutResamplingComboBox;
+  mMaximumOversamplingSpinBox = maximumOversamplingSpinBox;
+  mCbEarlyResampling = cbEarlyResampling;
+
+  mZoomedInResamplingComboBox->insertItem( 0, QObject::tr( "Nearest neighbour" ) );
```

Comment:
The nicer way to handle this kind of thing is to set the user data on the item to the enum value, and then use that to retrieve /set the combobox value. See .e.g 

https://github.com/qgis/QGIS/blob/master/src/gui/layout/qgslayouthtmlwidget.cpp#L62
https://github.com/qgis/QGIS/blob/master/src/gui/layout/qgslayouthtmlwidget.cpp#L182
https://github.com/qgis/QGIS/blob/master/src/gui/layout/qgslayouthtmlwidget.cpp#L435

Reply:
fixed